x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -342,13 +356,28 @@ export class FleetPlugin
 
     const fleetSetupPromise = (async () => {
       try {
+        this.fleetStatus$.next({
+          level: ServiceStatusLevels.degraded,
+          summary: 'Fleet is setting up',
+        });
+
         await setupFleet(
           new SavedObjectsClient(core.savedObjects.createInternalRepository()),
           core.elasticsearch.client.asInternalUser
         );
+
+        this.fleetStatus$.next({
+          level: ServiceStatusLevels.available,
+          summary: 'Fleet is available',
+        });
       } catch (error) {
         logger.warn('Fleet setup failed');
         logger.warn(error);
+
+        this.fleetStatus$.next({
+          level: ServiceStatusLevels.unavailable,
+          summary: 'Fleet setup failed',
```

Comment:
Can we add the details of the error message to the `description` or `meta` fields?

Reply:
Added `error.message` to `meta` here, e.g when package registry is unreachable:

![image](https://user-images.githubusercontent.com/6766512/144255010-7f68eb6b-c770-4ba5-b4f0-370bc229b01f.png)